From 1086e588db65e43d3919b72185a2cb60601fdd96 Mon Sep 17 00:00:00 2001
From: Huiquan Zhong <huiquan.zhong@intel.com>
Date: Fri, 12 Dec 2014 13:14:39 +0800
Subject: [PATCH] i2c: designware-pci: dw_i2c_init_driver as fs_initcall

Improve module init level to fs_initcall to prevent i2c register
adapter conflict since GPU i2c controller probe before us.

Also as I2C is a bus driver, it should be available as early as
possible.

Change-Id: Ieab92c392719098f319fa2517401ffabf7fb17ba
Signed-off-by: Huiquan Zhong <huiquan.zhong@intel.com>
(cherry picked from commit 8c29d6cdca85ca01c0dfee497551563cceda1397)

Signed-off-by: Brett T. Warden <brett.t.warden@intel.com>
---
 drivers/i2c/busses/i2c-designware-pcidrv.c |   12 +++++++++++-
 1 file changed, 11 insertions(+), 1 deletion(-)

diff --git a/drivers/i2c/busses/i2c-designware-pcidrv.c b/drivers/i2c/busses/i2c-designware-pcidrv.c
index f6ed06c..3ef3243 100644
--- a/drivers/i2c/busses/i2c-designware-pcidrv.c
+++ b/drivers/i2c/busses/i2c-designware-pcidrv.c
@@ -335,7 +335,17 @@ static struct pci_driver dw_i2c_driver = {
 	},
 };
 
-module_pci_driver(dw_i2c_driver);
+static int __init dw_i2c_init_driver(void)
+{
+	return pci_register_driver(&dw_i2c_driver);
+}
+fs_initcall(dw_i2c_init_driver);
+
+static void __exit dw_i2c_exit_driver(void)
+{
+	pci_unregister_driver(&dw_i2c_driver);
+}
+module_exit(dw_i2c_exit_driver);
 
 MODULE_AUTHOR("Baruch Siach <baruch@tkos.co.il>");
 MODULE_DESCRIPTION("Synopsys DesignWare PCI I2C bus adapter");
-- 
1.7.9.5

